docs/using/security-non-events.md

```diff
@@ -1,24 +1,35 @@
+This page lists security vulnerabilities which Sandstorm mitigated, such that Sandstorm users were never vulnerable to the bug&mdash;even before it was fixed.
 
-A few notes first:
+Most of the bugs listed below are publicly-disclosed security vulnerabilities against apps which have Sandstorm ports. Of course, the disclosures were made against the non-Sandstorm version of the app. On Sandstorm, these bugs either didn't matter at all, or their impact was drastically reduced.
+
+We also list some bugs below reported against the Linux kernel. These are bugs that normally would allow any process executing on a Linux machine to gain full control of the machine. Under a naive containerization environment, these bugs might allow an app to escape its container. However, Sandstorm's hardened sandbox protected against these vulnerabilities.
+
+## Background
+
+A few notes, to be sure we are on the same page.
 
 * Security is risk-management, not binary. No software, Sandstorm included, will ever protect all user data from all bugs in all programs. But, raising barriers to a successful attack means fewer successful attacks will occur, which is obviously valuable. Sandstorm aims to reduce the risk of attack by an order of magnitude or more.
 * Most real-world attacks do not involve novel new techniques or genius insights. Instead, attackers exploit the sad reality that most developers simply don't think carefully about their code. Developers often concern themselves primarily with making their code work in the common case; few developers ask themselves: "What is every possible path this code could take?" As a result, most apps are simply full of security bugs, and anyone looking for bugs in a particular app will find many in short order. Sandstorm therefore aims to (1) mitigate common app bugs, and (2) confine the damage caused by bugs it can't mitigate.
-* Of course, Sandstorm itself could be buggy, but Sandstorm is written by security nerds -- the kind of people who do ask themselves "What is every possible path this code could take?" Moreover, we plan to commission an outside security review before version 1.0. Hopefully, the chance of bugs in Sandstorm is much less than the chance of bugs in an average app.
-* The list below is not even remotely complete. Rather, it is a sample of the few bugs we've actually researched -- from the few apps that are big enough to have attracted security review and publicly disclosed vulnerabilities.
+* Sandstorm is NOT a "web application firewall". WAFs monitor for and block known attacks and suspicious behavior. Sandstorm uses fine-grained containerization and access control to mitigate vulnerabilities across the board, including vulnerabilities that haven't been discovered yet. See [how Sandstorm works](https://sandstorm.io/how-it-works) and [Sandstorm's security practices](security-practices) for details.
```

Comment:
One way to misunderstand Sandstorm while believing only true things is that "Sandstorm is a web application firewall that adds HTTP firewall rules that amount to access control checks." I agree that's not really the point of a web app firewall, though.